src/main/java/org/dependencytrack/event/EventSubsystemInitializer.java

```diff
@@ -87,11 +87,18 @@ public void contextInitialized(final ServletContextEvent event) {
         if (RequirementsVerifier.failedValidation()) {
             return;
         }
-        if (Config.getInstance().getPropertyAsBoolean(ConfigKey.BOM_PROCESSING_TASK_V2_ENABLED)) {
-            EVENT_SERVICE.subscribe(BomUploadEvent.class, BomUploadProcessingTaskV2.class);
-        } else {
-            EVENT_SERVICE.subscribe(BomUploadEvent.class, BomUploadProcessingTask.class);
+
+        //EXPERIMENTAL: FUTURE RELEASES SHOULD JUST REMOVE THE FOLLOWING BLOCK AND ENABLE THE COMMENT CODE
+        try (QueryManager qm = new QueryManager()) {
+            if (qm.isEnabled(ConfigPropertyConstants.BOM_PROCESSING_TASK_V2_ENABLED)) {
+                EVENT_SERVICE.subscribe(BomUploadEvent.class, BomUploadProcessingTaskV2.class);
+            } else {
+                EVENT_SERVICE.subscribe(BomUploadEvent.class, BomUploadProcessingTask.class);
+            }
```

Comment:
I feel like adding this check to `BomUploadProcessingTask`, and conditionally delegating to `BomUploadProcessingTaskV2` would be both less disruptive (no risk of losing events), and simpler (only one code area to check vs. two).

Reply:
Sorry didn't really get what you are advising

Reply:
What I'm suggesting is something like this:

```java
class BomUploadProcessingTask {
    void inform(Event event) {
        if (qm.isEnabled(BOM_UPLOAD_PROCESSING_TASK_V2_ENABLED)) {
            new BomUploadProcessingTaskV2().inform(event);
            return;
        }
        
        // ...
    }
}
```

Reply:
ok got it, but maybe the other way around... put this code on BomUploadProcessingTaskV2, and declare the subscription against it... this way if we want to remove v1 we just need to remove the if